### PR TITLE
Url option

### LINKED ISSUE #497: dnf download `--url`
Additional resources:

DNF4 implementation - https://github.com/rpm-software-management/dnf-plugins-core/blob/master/plugins/download.py (between lines 112 and 125)

Requirements:
Each package will be represented only by one url

Problems or recommendations:
RPM location could contain the whole url (then don't merge it with url of mirror)
Skip mirrors with unsupported protocols by Curl (rsync).
Metalinks and mirrorlists should be taken as primary source for urls - DNF5 use that approach also.

### PR DESCRIPTION
**Pull request by derickdiaz**
Fixes #497

Changes get the urls for the package specified by the --url option and returns the first url found as discussed.

### CODE REVIEW

File: dnf5/commands/download/download.cpp [hunk 1]
```diff
@@ -146,20 +155,34 @@ void DownloadCommand::run() {
...
 
-        for (auto & [nevra, pkg] : download_pkgs) {
-            downloader.add(pkg);
+    if (url_option->get_value()) {
+        for (auto & [nerva, pkg] : download_pkgs) {
+            auto urls = pkg.get_remote_locations();
+            if (urls.empty()) {
+                std::cout << "Failed to get mirror for package: " << pkg.get_name() << std::endl;
+                continue;
+            }
```
Comment (kontura) [inline dnf5/commands/download/download.cpp:168]: I don't think this should go to `stdout`. Either just log it like old dnf:
```cpp
ctx.base.get_logger()->warning("Failed to get mirror for package: \"{}\"", pkg.get_name());
```

or maybe even better we could try making it strict and throw an exception. 
```cpp
libdnf_assert(urls.empty(), "Failed to get mirror for package: \"{}\"", pkg.get_name());
```
I think this basically shouldn't happen but if it turns out to be a problem we can change it later.

**Review by kontura — CHANGES_REQUESTED** [review]
Thank you for working on this.

File: dnf5/commands/download/download.cpp (hunk 1 above)
Comment (derickdiaz) [reply dnf5/commands/download/download.cpp]: Used the libdnf_assert option as suggested

File: dnf5/commands/download/download.cpp [hunk 2]
```diff
@@ -146,20 +156,31 @@ void DownloadCommand::run() {
...
+    if (download_pkgs.empty()) {
+        return;
+    }
 
-        for (auto & [nevra, pkg] : download_pkgs) {
-            downloader.add(pkg);
+    if (url_option->get_value()) {
+        for (auto & [nerva, pkg] : download_pkgs) {
+            auto urls = pkg.get_remote_locations();
+            libdnf_assert(urls.empty(), "Failed to get mirror for package: \"{}\"", pkg.get_name());
```
Comment (kontura) [inline dnf5/commands/download/download.cpp:166]: Oh, the condition has to be reversed: `!urls.empty()`
This way it always fails.

**Review by kontura — CHANGES_REQUESTED** [review]
Can you squash the `Changed urloption to url ` commit into the first one?
We prefer to have the history as clean as possible.

File: dnf5/commands/download/download.cpp (hunk 2 above)
Comment (derickdiaz) [reply dnf5/commands/download/download.cpp:166]: No problem on the squash. I'm confused on the url.empty part. Wouldn't we want it to trigger an assert if the command when the vector is emptyl? Thank you

**Review by kontura — CHANGES_REQUESTED** [review]
Thank you, it looks good.
One last thing can you remove the `Changed urloption to url` and `reversed assertion condition` from the commit message? It doesn't really make sense when the commits are squashed.

**Review by kontura — APPROVED** [review]
Thank you!

**Pull request merged**